packages/components/src/ListBox/src/ListBoxItem.tsx

```diff
@@ -0,0 +1,313 @@
+import { CheckmarkIcon, IconContext, type IconProps } from "@hopper-ui/icons";
+import { type StyledComponentProps, useStyledSystem, type ResponsiveProp, useResponsiveValue } from "@hopper-ui/styled-system";
+import { forwardRef, type ReactNode, useContext, type ForwardedRef, useState, useEffect, useCallback } from "react";
+import {
+    useContextProps, 
+    ListBoxItem as RACListBoxItem, 
+    type ListBoxItemProps as RACListBoxItemProps, 
+    composeRenderProps,
+    DEFAULT_SLOT,
+    ListStateContext,
+    type ListBoxItemRenderProps
+} from "react-aria-components";
+
+import { BadgeContext } from "../../Badge/index.ts";
+import { Checkbox } from "../../checkbox/index.ts";
+import { IconListContext } from "../../IconList/index.ts";
+import { Radio, RadioGroup, RadioList } from "../../radio/index.ts";
+import { Text, TextContext, type TextProps } from "../../typography/Text/index.ts";
+import { composeClassnameRenderProps, type SizeAdapter, SlotProvider, cssModule, isTextOnlyChildren } from "../../utils/index.ts";
+
+import { ListBoxItemContext } from "./ListBoxItemContext.ts";
+
+import styles from "./ListBoxItem.module.css";
+
+export const GlobalListBoxItemCssSelector = "hop-ListBoxItem";
+
+export type ListBoxItemSize = "xs" | "sm" | "md" | "lg";
+
+export interface ListBoxItemProps<T> extends StyledComponentProps<Omit<RACListBoxItemProps<T>, "orientation | layout">> {
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size?: ResponsiveProp<ListBoxItemSize>;
+}
+
+interface ListBoxInnerProps extends ListBoxItemRenderProps {
```

Comment:
the listbox inner extends a listboxitem render props? it sounds weird, mixing listbox item props and listbox props

Reply:
the name of the interface is wrong, it should be ListBoxItemInnerProps

Reply:
right, I missed that one, sorry